introspection-engine/connectors/sql-introspection-connector/src/re_introspection.rs

```diff
@@ -135,61 +133,36 @@ pub fn enrich(old_data_model: &Datamodel, new_data_model: &mut Datamodel) -> Vec
-        for changed_model_name in &changed_model_names {
-            let changed_model = new_data_model.find_model(&changed_model_name.1.model).unwrap();
-
-            for rf in changed_model.relation_fields() {
-                let info = &rf.relation_info;
-                let other_model_in_relation = new_data_model.find_model(&info.to).unwrap();
-                let number_of_relations_to_other_model_in_relation = changed_model
-                    .relation_fields()
-                    .filter(|f| f.points_to_model(&info.to))
-                    .count();
-
-                let other_relation_field = new_data_model.find_related_field_bang(&rf);
-                let other_info = &other_relation_field.relation_info;
-
-                let (model_with_fk, referenced_model, fk_column_name) = if info.to_fields.is_empty() {
-                    // does not hold the fk
-                    (
-                        &other_model_in_relation.name,
-                        &changed_model.name,
-                        other_info.fields.join("_"),
-                    )
-                } else {
-                    // holds the fk
-                    (
-                        &changed_model.name,
-                        &other_model_in_relation.name,
-                        info.fields.join("_"),
-                    )
-                };
-
-                let unambiguous = number_of_relations_to_other_model_in_relation < 2;
-                let relation_name = if unambiguous {
-                    RelationNames::name_for_unambiguous_relation(model_with_fk, referenced_model)
-                } else {
-                    RelationNames::name_for_ambiguous_relation(model_with_fk, referenced_model, &fk_column_name)
-                };
-
-                relation_fields_to_change.push((changed_model.name.clone(), rf.name.clone(), relation_name.clone()));
-                relation_fields_to_change.push((
-                    other_model_in_relation.name.clone(),
-                    other_relation_field.name.clone(),
-                    relation_name.clone(),
-                ));
+        for model in new_data_model.models() {
+            for field in model.relation_fields() {
+                if let Some(old_model) = old_data_model.find_model(&model.name) {
+                    for old_field in old_model.relation_fields() {
+                        let related_field = &new_data_model.find_related_field_bang(&field);
+                        let old_related_field = &old_data_model.find_related_field_bang(&old_field);
+                        //the relationinfos of both sides need to be compared since the relationinfo of the
+                        // non-fk side does not contain enough information to uniquely identify the correct relationfield
+
+                        if &old_field.relation_info == &field.relation_info
+                            && old_related_field.relation_info == related_field.relation_info
```

Comment:
this works because of the custom `PartialEq` impl on `RelationInfo`, right?

Reply:
Yes, that one ignores the relation name for PartialEq